### DISCUSSION TITLE
Plans for Quartz .NET REST API?

### DISCUSSION THREAD

**Discussion by gburghardt** [quartznet/quartznet#1789] [Q&A]
I am evaluating Quartz, and we need some sort of UI for end users. They need to trigger jobs manually, and we certainly need some sort of output to visualize the state of all the jobs. I see there is a [Quartz.Web project](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Web), which has been gathering dust for years. I read #1597 and it mentioned the Quartz.Web project is just a proof of concept.

Looking at #1079, I see some 3rd party web UI's along with some commentary that they are feature-stable and haven't seen updates in quite a while as result. [Quartzmin](https://github.com/jlucansky/Quartzmin) looks promising, but runs on .NET Framework 4.5. Having this UI or REST API on .NET Core/Framework 6 is ideal, due to its long term support. Using .NET 4.x is not ideal, because Microsoft has started the sunsetting process for older versions of their framework.

I can see not building a full web UI. Front end technologies see a lot of churn, which becomes a maintenance headache. I would think a REST API would be easier to build into the main Quartz project. Two challenges would be authentication and authorization. I've also read through #511, but that conversation seems to have stalled out. Furthermore, a comment stated that authentication and authorization should be left out, which I found mind-boggling.

Adding authentication is easy in IIS. Just turn on active directory authentication. Authorization is a different issue, though. We are a large organization, and not everybody should be able to log in and manage these jobs. Using active directory groups is also not ideal, because these jobs are part of a larger application ecosystem that already has its own role/permission system.

I am reluctant to use this if I need to rely on a 3rd party to provide core functionality.

Are there any concrete plans for providing a REST API in the main Quartz repository? If so, is there a general timeline for implementation?

**Comment by lahma** [comment]
It would be great to have general REST API available in Quartz repository, but currently there is no timeline as I'm unaware of ongoing work.

**Reply by gburghardt** [reply]
I'm not against contributing. What do you think about at least starting to define the URL structure and maybe resultant data structures? Building a REST API can be a big project. Maybe we can take small bites? We would just need some sort of road map.

**Reply by lahma** [reply]
I think https://github.com/quartznet/quartznet/issues/511 is the best place to discuss and coordinate - like @Sicos1977 below noted he has work ongoing.

**Comment by Sicos1977** [comment]
I'm busy with a REST API --> https://github.com/Sicos1977/QuartzWebApi ... at the moment it is a slow process due to the lack of spare time. In a few week I have a vacation and planning to finish it. I think it is about 70% done. At the moment the project is just thrown together with code so I also need to cleanup the code.
